Add cloudberry-env.sh

leborchuk opened the pull request:
Add cloudberry-env.sh file and replacing greenplum-path.sh

The changes will be made in a future Apache Cloudberry release to
ensure compliance with Apache Software Foundation guidelines.

See details in 
- discussion https://github.com/apache/cloudberry/issues/1149
- mailing list https://lists.apache.org/thread/b8o974mnnqk6zpy86dgll2pgqcvqgnwm

Fixes #1149

tuhaihe commented:
For the file renaming, we need to replace all the strings related to `greenplum_path` in the source code completely, not just these three places. So it's a huge work and needs more care. Thanks @leborchuk!

tuhaihe commented:
> > For the file renaming, we need to replace all the strings related to `greenplum_path` in the source code completely, not just these three places. So it's a huge work and needs more care. Thanks @leborchuk!
> 
> Yep, I see 85 greenplum_path occurencies of greenplum_path in a source code
> 
> ```
> xifos@xifos-dev-jammy:~/git/cloudberry$ grep -l -r greenplum_path * | wc -l
> 85
> ```
> 
> What I am going to do is replace them in the next PR. I have already checked these PR in my development environment and wanted to move forward step by step.
> 
> Is it OK? (I still could replace all occurrences in one PR if needed)

‌Let's replace them all in this PR.

tuhaihe commented:
> Sorry for the noise here. The file generate-greenplum-path.sh has been changed, so I should rebase. I left generate-greenplume-path.sh as-is with the "NOTICE from the Apache Cloudberry PPMC".
> 
> I think we will delete it when all the work is done.

Since we're replacing all the old filenames in this PR, the NOTICE file becomes unnecessary. I recommend removing both the license headers and NOTICE references in this PR as well. @leborchuk FYI.

leborchuk commented:
> > Sorry for the noise here. The file generate-greenplum-path.sh has been changed, so I should rebase. I left generate-greenplume-path.sh as-is with the "NOTICE from the Apache Cloudberry PPMC".
> > I think we will delete it when all the work is done.
> 
> Since we're replacing all the old filenames in this PR, the NOTICE file becomes unnecessary. I recommend removing both the license headers and NOTICE references in this PR as well. @leborchuk FYI.

Yep, removed the NOTICE and all occurrences of greenplum_path that were found in the code. Some of them were manually executed scripts, like src/backend/gporca/scripts/get_debug_event_counters.py, or inside beehave tests. I just changed them using a sed script and didn't check if they still worked or not. I launched build&configure&create/destroy&check_world, and everything worked fine. I believe that should be enough )

tuhaihe commented:
> > > Sorry for the noise here. The file generate-greenplum-path.sh has been changed, so I should rebase. I left generate-greenplume-path.sh as-is with the "NOTICE from the Apache Cloudberry PPMC".
> > > I think we will delete it when all the work is done.
> > 
> > 
> > Since we're replacing all the old filenames in this PR, the NOTICE file becomes unnecessary. I recommend removing both the license headers and NOTICE references in this PR as well. @leborchuk FYI.
> 
> Yep, removed the NOTICE and all occurrences of greenplum_path that were found in the code. Some of them were manually executed scripts, like src/backend/gporca/scripts/get_debug_event_counters.py, or inside beehave tests. I just changed them using a sed script and didn't check if they still worked or not. I launched build&configure&create/destroy&check_world, and everything worked fine. I believe that should be enough )

Cool! Thanks @leborchuk, will check it later.

tuhaihe commented on gpMgmt/Makefile:
```diff
@@ -6,12 +6,13 @@ SUBDIRS= sbin bin doc
 $(recurse)
 
-generate_greenplum_path_file:
+generate_cloudberry_path_file:
```
We can also update the rule name to address the file name change:


```suggestion
generate_cloudberry_env_file:
```

tuhaihe commented on gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py:
```diff
@@ -2619,14 +2619,14 @@ def impl(context, location):
     """
     gphome = os.environ["GPHOME"]
-    greenplum_path = path.join(gphome, 'greenplum_path.sh')
+    cloudberry_path = path.join(gphome, 'cloudberry-env.sh')
```
That would be better to update this variable name too.

leborchuk replied:
Yep, we cannot use "-" in python, but the idea is clear, fixed

tuhaihe reviewed (changes requested):
@leborchuk thanks for your contribution. Just left a few comments, FYI.

tuhaihe commented on gpMgmt/Makefile:
```diff
@@ -6,12 +6,13 @@ SUBDIRS= sbin bin doc
 	unset LIBPATH; \
-	bin/generate-greenplum-path.sh > $(DESTDIR)$(prefix)/greenplum_path.sh
+	bin/generate-cloudberry-env.sh > $(DESTDIR)$(prefix)/cloudberry-env.sh
+	cp $(DESTDIR)$(prefix)/cloudberry-env.sh $(DESTDIR)$(prefix)/greenplum_path.sh
```
Here, we need to update the related files in the repo #https://github.com/apache/cloudberry-devops-release/ at first to make the test pass.

tuhaihe replied:
Hi, we need to update the following files under the apache/cloudberry-site repo:

For main branch:
- https://github.com/apache/cloudberry-site/blob/main/docs/deployment/quick-build.md
- https://github.com/apache/cloudberry-site/blob/main/docs/deployment/set-demo-cluster.md

For 2.x branch:
- https://github.com/apache/cloudberry-site/blob/main/versioned_docs/version-2.x/deployment/quick-build.md
- https://github.com/apache/cloudberry-site/blob/main/versioned_docs/version-2.x/deployment/set-demo-cluster.md

Thanks!

tuhaihe commented:
Also, I found we need to update the following line to pass the license check: 

https://github.com/apache/cloudberry/blob/6b968245a59fc4e26b0cd842412c33403738afe2/pom.xml#L304.

tuhaihe commented:
BTW, welcome to help review this PR: https://github.com/apache/cloudberry/pull/1171. We need to add the license check to the CI workflow.
